### DISCUSSION TITLE
Debugging tests written in Typescript

### DISCUSSION THREAD

**Discussion by pYr0x** [vitest-dev/vitest#488] [Q&A]
i wrote my unit tests in typescript. like 
```
test('true', () => {
  const test = true;
  expect(test).toBe(true);
})
```
but i can not debug them. 
i saw that `vitest` using `jest` but i didn't found out what is compiling typescript to javascript and if there are sourcemaps created? i added a `jest.config.js` with 
```
transform: {
    '^.+\\.(ts)$': 'ts-jest',
  },
```
but didn't help.
i using `Jetbrain` as IDE. In a "normal" jest environment with `ts-jest` i can debug my tests written in typescript.

**Comment by sheremet-va** [comment]
We don't use Jest. Im not sure if IDEA supports custom testing setup. Maybe someone can help you here?

**Reply by rockwotj** [reply]
You would need a custom intellij plugin for this: https://www.jetbrains.com/help/idea/unit-testing-javascript.html

**Reply by rockwotj** [reply]
Here the task to track vitest support for intellij https://youtrack.jetbrains.com/issue/WEB-54437
